core/src/main/scala/org/apache/spark/deploy/mesos/ui/MesosClusterPage.scala

```diff
@@ -68,6 +75,25 @@ private[mesos] class MesosClusterPage(parent: MesosClusterUI) extends WebUIPage(
 
   private def driverRow(state: MesosClusterSubmissionState): Seq[Node] = {
     val id = state.driverDescription.submissionId
+    val masterInfo = parent.scheduler.getSchedulerMasterInfo()
+    val schedulerFwId = parent.scheduler.getSchedulerFrameworkId()
+    val sandboxCol = if (masterInfo.isDefined && schedulerFwId.isDefined) {
+
+      val masterUri = masterInfo.map{info => s"http://${getIp4(info.getIp)}:${info.getPort}"}.get
+      val directory = getTaskDirectory(masterUri, id, state.slaveId.getValue)
+
+      if(directory.isDefined) {
+        val sandBoxUri = s"$masterUri" +
+          s"/#/slaves/${state.slaveId.getValue}" +
+          s"/browse?path=${directory.get}"
+          <a href={sandBoxUri}>Sandbox</a>
```

Comment:
Yes i know, but had now way to test it... dont really know the correct format there for the url... please provide the specs in our document.